### PR TITLE
feat: add useActiveDescendant hook to simplify child element focus in Carousel

### PR DESCRIPTION
**Pull request by wp-aberg**
## What I did

This PR adds a custom hook `useActiveDescendant` that helps to manage keyboard navigation and focus for child interactive elements in the Carousel 

```javascript
const { addDescendant, handleDescendantFocus, contentProps } =
    useActiveDescendant();
```
The descendent focus hander is added to the root to monitor the current item with descendant focus
```javascript
<Carousel.Root onDescendantFocus={handleDescendantFocus}>
```
Content props are expanded to pass event handlers and the current `aria-activedescendant` to the content element
```javascript
<Carousel.Content  {...contentProps}>
```
Each CarouselItem must have an id that is then used by its interactive children with the `addDescendant` helper in its ref callback. The interactive child elements must also have an id and have their tabIndex set to -1 to prevent them from stealing focus
```javascript
<Carousel.Item id={item.id}>
  <a
    href="#"
    tabIndex={-1}
    id={`${item.id}-link`}
    ref={(el) => {
      addDescendant({
        element: el,
        id: `${item.id}-link`,
        parentId: item.id,
      });
    }}
  >
  By The Washington Post
  </a>
</Carousel.Item>
```

SRED-289

### CODE REVIEW

File: ui/carousel/src/CarouselContent.tsx [hunk 1]
```diff
@@ -166,42 +166,47 @@ export const CarouselContent = React.forwardRef<
     preventScrollOnSwipe: true,
   });
 
-  const handleOnFocus = () => {
+  const handleOnFocus = (event) => {
```
Comment (ebgranger) [inline ui/carousel/src/CarouselContent.tsx:169]: I'm glad to see `event` and not `e` 🌈

File: ui/carousel/src/CarouselContent.tsx [hunk 2]
```diff
@@ -241,11 +246,12 @@ export const CarouselContent = React.forwardRef<
         <Slider css={{ transform: `translateX(${xPos}px)` }}>
           {React.Children.map(children, (child, index) => {
             if (React.isValidElement(child)) {
+              console.log();
```
Comment (ebgranger) [inline ui/carousel/src/CarouselContent.tsx:249]: blank log

Comment (ebgranger) [inline ui/carousel/src/CarouselContent.tsx:249]:
```suggestion
```

File: build.washingtonpost.com/docs/components/carousel.mdx [hunk 3]
```diff
@@ -912,50 +912,60 @@ Add `aria-labelledby` to Carousel Items to allow the screen-reader to read the c
...
     {
+      id: "story-6",
       kicker: "THE FIX",
       headline: "Alito’s remarkable reasoning",
       blurb:
         "Alito suggested the Biden administration might simply disregard an unfavorable ruling on mifepristone. Here's what he appears to be talking about.",
     },
   ];
+
+  const { addDescendant, contentProps, handleDescendantFocus } =
```
Comment (artmsilva) [inline build.washingtonpost.com/docs/components/carousel.mdx:958]: Maybe we could have a new example with a blurb explaining how to use this like "Links in Carousels". Good idea for a new ticket

**Review by ebgranger — APPROVED** [review]
looks great, testing was successful.

I think tests would be essential for something of this complexity but we do those on Monday.

**Pull request merged**